integreat_cms/cms/views/regions/region_actions.py

```diff
@@ -45,25 +46,42 @@ def delete_region(request, *args, **kwargs):
     # Prevent ProtectedError when location gets deleted before their events
     region.events.update(location=None)
     # Delete region and cascade delete all contents
-    deleted_objects = region.delete()
-    logger.info(
-        "%r deleted %r, cascade deleted objects: %r",
-        request.user,
-        region,
-        deleted_objects,
-    )
-    # Get orphan users who aren't superuser or staff and don't have a region assigned
-    # (Creating users with these combination is impossible, so they were region users of the deleted region before)
-    orphan_users = get_user_model().objects.filter(
-        is_superuser=False, is_staff=False, regions=None
-    )
-    if orphan_users.exists():
+    try:
+        deleted_objects = region.delete()
         logger.info(
-            "Deleted orphan users: %r",
-            orphan_users,
+            "%r deleted %r, cascade deleted objects: %r",
+            request.user,
+            region,
+            deleted_objects,
         )
-        orphan_users.delete()
+        # Get orphan users who aren't superuser or staff and don't have a region assigned
+        # (Creating users with these combination is impossible, so they were region users of the deleted region before)
+        orphan_users = get_user_model().objects.filter(
+            is_superuser=False, is_staff=False, regions=None
+        )
+        if orphan_users.exists():
+            logger.info(
+                "Deleted orphan users: %r",
+                orphan_users,
+            )
+            orphan_users.delete()
 
-    messages.success(request, _("Region was successfully deleted"))
+        messages.success(request, _("Region was successfully deleted"))
+    except ProtectedError:
+        mirrored_pages = []
+        pages = Page.objects.all()
+        for page in pages:
+            mirrored_pages.append(page.mirrored_page)
+        p = ""
+        for page in mirrored_pages:
+            if page is not None:
+                if page.region == region:
+                    p = p + str(page)
+        messages.error(
+            request,
+            _(
+                "Region could not be deleted, because these pages are {} mirrored in other regions"
+            ).format(p),
+        )
```

Comment:
I think this can be simplified:

```suggestion
    except ProtectedError:
        pages = Page.objects.filter(region=region)
        messages.error(
            request,
            _(
                "Region could not be deleted, because the following page(s) are mirrored in other regions: {}"
            ).format(", ".join([str(page) for page in pages])),
        )
```

Reply:
I agree on using the `join` instead of the loop, but this lists just all pages of the region and not those which are embedded in other regions as live content...

Reply:
Ah, you're absolutely right, I misunderstood what was being done & had an unfortunate test case.